examples/nextjs-todo/pages/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState } from "react";
+import { withAuth } from "@/server/withAuth";
+import { GetServerSideProps } from "next";
+
+import { getClient } from "@/arke/getClient";
+import useClient from "@/arke/useClient";
+import { Client, TUnit } from "@arkejs/client";
+
+import TodoForm from "../components/ui/TodoForm";
+import Todo from "@/components/ui/Todo";
+import { Button } from "@arkejs/ui";
+
+const ArkeTodoTitle = () => {
+  return (
+    <>
+      <div className="flex items-center space-x-3">
+        <div className="w-6 h-6 border-y-[3px] border-black flex justify-center items-center">
+          <div className="w-1.5 h-1.5 bg-black rounded-full" />
+        </div>
+        <h1 className="text-3xl font-bold">ARKE todo</h1>
+      </div>
+    </>
+  );
+};
+
+export default function ArkeTodo({ todos }: { todos: TUnit[] }) {
+  const [allTodos, setAllTodos] = useState(todos);
+  const [modal, onSetModal] = useState({ isOpen: false, id: null });
+
+  const client = useClient();
+
+  async function getTodosData() {
+    const response = await fetchAllTodos(client);
+    setAllTodos(response.data.content.items);
+  }
+
+  const handleCreate = async (data: TUnit, getData: () => {}) => {
+    client.unit
+      .create("todo", data)
+      .then((res) => getTodosData())
+      .catch((e) => console.log("something went wrong", e));
+  };
+
+  const handleEdit = async (data: TUnit, id: string, getData: () => {}) => {
+    client.unit
+      .edit("todo", id, data)
+      .then((res) => getTodosData())
+      .catch((e) => console.log("something went wrong ", e));
+  };
+
+  return (
+    <>
+      <div className="relative p-12 h-screen">
+        <header className="flex justify-between items-center space-x-2">
+          <ArkeTodoTitle />
+          <Button
+            className="btn--secondary rounded-full"
+            onClick={() => onSetModal({ ...modal, isOpen: true })}
+          >
+            Add todo
+          </Button>
+        </header>
+        <main className="py-8">
+          <div className="relative pt-2">
+            <h1 className="absolute -top-1 px-4 bg-black text-white rounded-full z-50">
+              Todos
+            </h1>
+            <ul className="relative flex space-x-6 overflow-y-auto pt-6 pb-8 border-t-2 border-black">
+              {allTodos
+                .filter((item) => item.done === false)
+                .map((item) => (
+                  <li key={item.id}>
+                    <Todo
+                      title={item.title}
+                      description={item.description}
+                      id={item.id}
+                      done={item.done}
+                      getTodosData={getTodosData}
+                      onClose={onSetModal}
+                      onSubmit={modal}
+                    />
```

Comment:
- onClose should call `setActiveTodoId` with `undefined`
- onSubmit should call `setActiveTodoId` with `undefined` and `getTodosData`